lib/IRGen/GenFunc.cpp

```diff
@@ -625,22 +625,25 @@ static void emitApplyArgument(IRGenFunction &IGF,
                         silConv.getSILType(substParam, substFnTy), in, out);
 }
 
-static CanType getArgumentLoweringType(CanType type,
-                                       SILParameterInfo paramInfo) {
+static CanType getArgumentLoweringType(SILType type,
+                                       SILParameterInfo paramInfo,
+                                       const SILFunction& SILFn,
+				       bool isNoEscape) {
+  CanType canType = type.getASTType();
   switch (paramInfo.getConvention()) {
   // Capture value parameters by value, consuming them.
   case ParameterConvention::Direct_Owned:
   case ParameterConvention::Direct_Unowned:
   case ParameterConvention::Direct_Guaranteed:
   case ParameterConvention::Indirect_In:
   case ParameterConvention::Indirect_In_Constant:
+    return canType;
   case ParameterConvention::Indirect_In_Guaranteed:
-    return type;
-
+    return type.isLoadable(SILFn) || !isNoEscape ? canType : CanInOutType::get(canType);  
```

Comment:
This line looks a little bit too long. Maybe use an 'if' instead of ternary operator